analyses/cell-type-ewings/cell_type_annotations/SCPCS000492/SCPCL000824_tumor-gene-expression-exploration.Rmd

```diff
@@ -0,0 +1,670 @@
+
+```{r}
+# pull out threshold used for assigning cells from ref sample 
+ref_threshold <- ref_assignment$markers$aucThr$selected
+
+# create a vector of tumor cells in 824 using the cutoff from 822
+new_tumor_cells <- new_auc@assays@data$AUC |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("gene_set") |> 
+  tidyr::pivot_longer(!gene_set, 
+                      names_to = "barcodes", 
+                      values_to = "auc") |> 
+  dplyr::filter(gene_set == "markers", auc >= ref_threshold) |>
+  dplyr::pull(barcodes)
+
+# add to original classification df 
+classification_df <- classification_df |> 
+  dplyr::mutate(auc_classification = dplyr::if_else(barcodes %in% new_tumor_cells, "Tumor", "Normal"))
+```
+
+```{r}
+# visualize which cells are classified as tumor 
+ggplot(classification_df, aes(x = UMAP1, y = UMAP2, color = auc_classification)) +
+  geom_point(alpha = 0.5, size = 0.5)
+```
+
+This looks similar to using the updated marker gene cutoff to classify cells where most cells in that bottom group are tumor cells.  
+
+## Classifying tumor cells with UCell
+
+Here we will look at using [`UCell`](https://carmonalab.github.io/UCell_demo/UCell_matrix_vignette.html#content) to calculate a gene set score for each cell and then attempt to classify cells based on the distribution of those gene set scores. 
+Again, we will look at both `SCPCL000822` and `SCPCL000824`. 
+
+```{r}
+# create list to use for ucell of gene sets and marker genes
+ucell_gene_sets <- c(genes_list, markers = list(marker_genes))
+
+# run ucell on both ref (822) and 824
+ref_ucell <- UCell::ScoreSignatures_UCell(counts(ref_sce), features = ucell_gene_sets)
+new_ucell <- UCell::ScoreSignatures_UCell(counts(sce), features = ucell_gene_sets)
+```
+
+`UCell` returns a signature score for each gene set and does not calculate any thresholds for classification on it's own. 
+We will look at the distribution of scores across both samples below and see if we can identify a good cut off to use for classification. 
+
+```{r}
+# plot distribution of ucell scores for both samples 
+ref_ucell <- ref_ucell |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("barcodes") |> 
+  dplyr::mutate(sample = "SCPCL000822")
+  
+ucell_df <- new_ucell |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("barcodes") |> 
+  dplyr::mutate(sample = "SCPCL000824") |> 
+  dplyr::bind_rows(ref_ucell) |> 
+  tidyr::pivot_longer(ends_with("UCell"), 
+                      names_to = "gene_list", 
+                      values_to = "signature_score") |> 
+  dplyr::mutate(
+    gene_list = stringr::word(gene_list, 1, sep = "_")
+  )
+```
+
+```{r}
+ggplot(ucell_df, aes(x = signature_score, colour = sample)) +
+  geom_density() +
+  facet_grid(rows = vars(gene_list))
+```
+
+Here we see that the marker gene set maybe has a small bimodal distribution in `SCPCL000822`, but it is very small and not quite as striking as looking at the scores from `AUCell` or raw gene expression. 
```

Comment:
I think this is an artifact of scaling. I would make set free y scales for easier comparisons here. To my squinty eye it actually looks like you are getting better results with the markers than any of the other gene sets.